### PR TITLE
Upgrade npm packages

### PR DESCRIPTION
**Pull request by dishanfernando91**
## Description

PR contains the following:

- Update outdated npm packages
- Replaced `node-sass` (deprecated) with `sass`
- Migrate to Vite

image

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [x] Code refactor
- [ ] Performance improvement
- [ ] Test improvements

## Related Issues

N/A

## How Has This Been Tested?

Tested on local. 

@joebeernink I did not see anything break, but I did not compare with the figma designs to check for any UI misalignments. 

## Notes

- Several deprecation warnings related to node-sass implementation. I've ignored them and hope that CSS changes will resolve those automatically.

image

### CODE REVIEW

File: TrashMob/client-app/package.json [hunk 1]
```diff
@@ -40,13 +42,17 @@
...
   },
   "scripts": {
-    "start": "react-scripts start",
-    "build": "react-scripts build",
-    "test": "react-scripts test",
-    "eject": "react-scripts eject"
+    "start": "vite",
+    "build": "vite build",
+    "test": "vite test",
+    "eject": "vite eject"
```
Comment (pitipatdop) [inline TrashMob/client-app/package.json:55]: There's no `vite eject`, can remove this command.

**Review by pitipatdop — APPROVED** [review]
LGTM!

Just one minor comment, we can remove `vite reject` command. Other than that, looks good.

File: TrashMob/client-app/package.json (hunk 1 above)
Comment (dishanfernando91) [reply TrashMob/client-app/package.json]: Removed. Thanks.

**Comment by joebeernink** [comment]
Error in the build:
  npm error code EJSONPARSE
  npm error path D:\a\TrashMob\TrashMob\TrashMob\client-app/package.json
  npm error JSON.parse Unexpected token "}" (0x7D) in JSON at position 1702 while parsing near "...t\": \"vite test\",\r\n  },\r\n  \"eslintConfig\"..."
  npm error JSON.parse Failed to parse JSON data.
  npm error JSON.parse Note: package.json must be actual JSON, not just JavaScript.

**Comment by pitipatdop** [comment]
There's an extra comma in package.json file, I removed it.

**Pull request merged**